header-verifier/tests/header_verifier_blackbox_test.rs

```diff
@@ -110,46 +110,57 @@ impl HeaderVerifierTestState {
             .run();
     }
 
-    fn propose_remove_execute_hash(
+    fn propose_remove_execute_hash_wrong_caller(
         &mut self,
         hash_of_hashes: &ManagedBuffer<StaticApi>,
         operation_hash: ManagedBuffer<StaticApi>,
+    ) {
+        self.world
+            .tx()
+            .from(OWNER)
+            .to(HEADER_VERIFIER_ADDRESS)
+            .typed(header_verifier_proxy::HeaderverifierProxy)
+            .remove_executed_hash(hash_of_hashes, operation_hash)
+            .returns(ExpectError(
+                4,
+                "Only ESDT Safe contract can call this endpoint",
+            ))
+            .run()
+    }
+
+    fn propose_remove_executed_hash(
+        &mut self,
+        caller: TestSCAddress,
+        hash_of_hashes: &ManagedBuffer<StaticApi>,
+        operation_hash: ManagedBuffer<StaticApi>,
         error_status: Option<ErrorStatus>,
     ) {
+        let transaction = self
+            .world
+            .tx()
+            .from(caller)
+            .to(HEADER_VERIFIER_ADDRESS)
+            .typed(header_verifier_proxy::HeaderverifierProxy)
+            .remove_executed_hash(hash_of_hashes, operation_hash);
+
         match error_status {
-            Some(error) => self
-                .world
-                .tx()
-                .from(OWNER)
-                .to(HEADER_VERIFIER_ADDRESS)
-                .typed(header_verifier_proxy::HeaderverifierProxy)
-                .remove_executed_hash(hash_of_hashes, operation_hash)
+            Some(error) => transaction
                 .returns(ExpectError(error.code, error.error_message))
                 .run(),
-            None => self
-                .world
-                .tx()
-                .from(ENSHRINE_ADDRESS)
-                .to(HEADER_VERIFIER_ADDRESS)
-                .typed(header_verifier_proxy::HeaderverifierProxy)
-                .remove_executed_hash(hash_of_hashes, operation_hash)
-                .run(),
+            None => transaction.run(),
         }
     }
-
     fn get_bls_keys(&mut self, bls_keys_vec: Vec<ManagedBuffer<StaticApi>>) -> BlsKeys {
-        let bls_keys = bls_keys_vec.iter().map(|key| key.clone().into()).collect();
+        let bls_keys = bls_keys_vec.iter().map(|key| key.clone()).collect();
 
         bls_keys
     }
 
-    fn get_mock_operation(
+    fn generate_bridge_operation_struct(
```

Comment:
You change this function but like this is not reusable multiple times.
Just make one function that generates the operations and another one for checking maybe and use both of them in tests.